src/test/java/org/broadinstitute/hellbender/tools/exome/allelefraction/AFModelUnitTest.java

```diff
@@ -0,0 +1,246 @@
+                }
+            }
+        }
+    }
+
+    @Test
+    public void testMCMC() {
+        final int numSamples = 1000;
+        final int numBurnIn = 200;
+
+        final double averageHetsPerSegment = 20;
+        final int numSegments = 100;
+        final int averageDepth = 50;
+
+        final double meanBias = 1.1;
+        final double biasRate = 100;
+        final double pOutlier = 0.02;
+
+        // note: the following tolerances could actually be made much smaller if we used more segments and/or
+        // more hets -- most of the error is the sampling error of a finite simulated data set, not numerical error of MCMC
+        final double minorFractionTolerance = 0.03;
+        final double meanBiasTolerance = 0.03;
+        final double biasRateTolerance = 500;
+        final double pOutlierTolerance = 0.03;
+        AFSimulatedData truthData = new AFSimulatedData(averageHetsPerSegment, numSegments,
+                averageDepth, meanBias, biasRate, pOutlier);
+
+
+        final AFModel model = new AFModel(truthData.getData());
+        model.fitMCMC(numSamples, numBurnIn);
+
+
+        final List<Double> meanBiasSamples = model.getmeanBiasSamples();
+        Assert.assertEquals(meanBiasSamples.size(), numSamples);
+
+        final List<Double> biasRateSamples = model.getBiasRateSamples();
+        Assert.assertEquals(biasRateSamples.size(), numSamples);
+
+        final List<Double> pOutlierSamples = model.getOutlierProbabilitySamples();
+        Assert.assertEquals(pOutlierSamples.size(), numSamples);
+
+        final List<AFState.MinorFractions> minorFractionsSamples = model.getMinorFractionsSamples();
+        Assert.assertEquals(minorFractionsSamples.size(), numSamples);
+        for (final AFState.MinorFractions sample : minorFractionsSamples) {
+            Assert.assertEquals(sample.size(), numSegments);
+        }
+
+        final List<List<Double>> minorFractionsSamplesBySegment = model.getMinorFractionSamplesBySegment();
+
+        final double MCMCMeanBias = meanBiasSamples.stream().mapToDouble(x -> x).average().getAsDouble();
+        final double MCMCBiasRate = biasRateSamples.stream().mapToDouble(x -> x).average().getAsDouble();
+        final double MCMCPOutlier = pOutlierSamples.stream().mapToDouble(x -> x).average().getAsDouble();
+        final List<Double> MCMCMinorFractions = minorFractionsSamplesBySegment
+                .stream().map(list -> list.stream().mapToDouble(x -> x).average().getAsDouble())
+                .collect(Collectors.toList());
+
+        double totalSegmentError = 0.0;
+        for (int segment = 0; segment < numSegments; segment++) {
+            totalSegmentError += Math.abs(MCMCMinorFractions.get(segment) - truthData.getTrueState().minorFractionInSegment(segment));
+        }
+
+        Assert.assertEquals(MCMCMeanBias, meanBias, meanBiasTolerance);
+        //Assert.assertEquals(MCMCBiasRate, biasRate, biasRateTolerance);
```

Comment:
The model is so insensitive to the bias rate that the posterior is nearly flat.  The only way to ensure passing a test with a non-trivial tolerance would be to increase the number of samples.